src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -992,6 +1002,280 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_JoinerStart(voi
     return error;
 }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
+
+static_assert(OPENTHREAD_API_VERSION >= 72, "SRP Client requires a more recent OpenThread version");
+
+template <class ImplClass>
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::OnSrpClientNotification(otError aError,
+                                                                                  const otSrpClientHostInfo * aHostInfo,
+                                                                                  const otSrpClientService * aServices,
+                                                                                  const otSrpClientService * aRemovedServices,
+                                                                                  void * aContext)
+{
+    switch (aError)
+    {
+    case OT_ERROR_NONE: {
+        ChipLogProgress(DeviceLayer, "OnSrpClientNotification: Last requested operation completed successfully");
+
+        if (aRemovedServices)
+        {
+            otSrpClientService * service = const_cast<otSrpClientService *>(aRemovedServices);
+            otSrpClientService * next    = nullptr;
+
+            // Free memory for all removed services.
+            do
+            {
+                next = service->mNext;
+                // Clear also names associated with the SrpClient::Service object
+                memset(const_cast<char *>(service->mInstanceName), 0, strlen(service->mInstanceName) + 1);
```

Comment:
How about you simply assign `service->mInstanceName = nullptr` or `memset(service, 0, sizeof(otSrpClientService));` and in the function looking for a free slot you compare `mService.mInstanceName`? Should be a little bit more efficient and will the outcome should be the same.

Reply:
I thought about it, but I decided on current solution, as it's more transparent when reading code. It we do as you suggest, we will meet a situation when for some time Service has assigned mName and mInstanceName, but actually otSrpClientService does not exist. It will not lead to any faults for now, but I imagine that someone will try to edit code in the future and will be pretty much confused that data are not consistent.

Reply:
Hmm I'm also not convinced.. but If we need to clear this, maybe just `service->mInstanceName[0] = '\0'`, same with `mName`?